### DISCUSSION TITLE
Simulink realtime + casadi

### DISCUSSION THREAD

**Discussion by RickDW** [casadi/casadi#3185] [Q&A] [answered]
Hi folks! I'm working on a project that requires Casadi to be used in a Simulink block. I've tried to work through the examples on the Casadi blog (i.e. https://web.casadi.org/blog/mpc-simulink/ and https://web.casadi.org/blog/mpc-simulink2/) but with this approach it is not possible to use code generation, which is required for Simulink Realtime. 
My current plan is to use Casadi's built-in code generation functionality to generate C code, and then include that in a separate block in the Simulink diagram. I'm not sure if this is the best approach, or if it'll work at all, but it's the best option I have right now. So I was wondering if there's anyone here who has run into the same problem, and if you might want to point me in the right direction :)

P.S. The code generation error that I ran into in Simulink is this one:

Compilation of model '[mpctest](matlab:open_system ('mpctest'))' failed while trying to resolve underspecified signal dimensions.
Caused by:
Simulink cannot automatically infer the output signal properties of 'mpctest/Subsystem/MATLAB System'. Simulink uses code generation technology to automatically determine the output signal properties from the System object.  The System object 'casadi_block' contains code that does not support code generation.

 The error is 
 'Overloading the method 'subsref' is not supported for code generation.'

 The error occurred for MATLAB System block ['mpctest/Subsystem/MATLAB System'](matlab:hilite(get_param('mpctest/Subsystem/MATLAB System','object'));). See line 53, column 5 in file ['/home/rick/Museum/applications/casadi-3.6.2-linux64-matlab2018b/+casadi/GenericMatrixCommon.m'](matlab:opentoline('/home/rick/Museum/applications/casadi-3.6.2-linux64-matlab2018b/+casadi/GenericMatrixCommon.m',53,5)). The error was detected during size propagation phase.
 Start [code generation report](matlab: cgxeprivate('InferenceReportDisplay','mpctest/Subsystem/MATLAB System');). 

 To prevent this error, use one of the following:
 * Modify the System object to avoid code that does not support [code generation](matlab:helpview(fullfile(docroot,'simulink','helptargets.map'),'blockCodeGeneration')).
 * Implement [propagation methods](matlab:helpview(fullfile(docroot,'simulink','helptargets.map'),'mixinPropagates')) in the System object code.

**Comment by jgillis** [comment, marked as answer]
Hi @RickDW , you probably missed this one: https://web.casadi.org/blog/s-function/
